### ISSUE TITLE
Vim9: fnamemodify() which the first argument is empty string cause "E928: String required"

### ISSUE THREAD

**Issue by rbtnn** [vim/vim#7465]
**Describe the bug**

**To Reproduce**
1. execute following codes

__codes__
```
vim9script
call fnamemodify('', '')
```

__output__
```
"Desktop\a.vim" [unix] 2L, 38B written
line    2:
E928: String required
```

**Expected behavior**
No error

**Environment:**
 - Vim version 8.2.2129

**Comment by lacygoill** [comment]
I noticed this too in https://github.com/vim/vim/issues/7447 , but I closed the issue because I think it's working as intended.  See this test:

https://github.com/vim/vim/blob/0ea7421ae6618fe8efe9e3afc8fdcf7a4d89739a/src/testdir/test_vim9_builtin.vim#L283

This change comes from [8.2.2117](https://github.com/vim/vim/releases/tag/v8.2.2117).

I think the rationale is that passing an empty string to `fnamemodify()` is useless, and probably caused by some mistake.  Instead of ignoring this mistake, which could have consequences much later in the code, and thus be hard to debug, Vim tries to be helpful and raise an error immediately to let the user know that something might be wrong in their code.  Could be wrong on my interpretation, but the test shows that the error is by design, and not by accident.

**Comment by rbtnn** [comment]
@lacygoill  Thanks you for your comments. I did not notice [this](https://github.com/vim/vim/blob/0ea7421ae6618fe8efe9e3afc8fdcf7a4d89739a/src/testdir/test_vim9_builtin.vim#L283). 
I think that it should change E928 to new error(e.g. `E999: non-empty string required`) because we can not guess that this means "type error" but "non-empty string required".

**Comment by brammool** [comment]
fixed by 8.2.2133
